controllers/operatorscaler_controller.go

```diff
@@ -0,0 +1,364 @@
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return false
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return false
+		},
+	}
+)
+
+type OperatorScalerReconciler struct {
+	client.Client
+	Scheme            *runtime.Scheme
+	OperatorNamespace string
+	ctx               context.Context
+	log               logr.Logger
+	cache             cache.Cache
+	controller        controller.Controller
+	watching          map[string]bool
+}
+
+//+kubebuilder:rbac:groups=apiextensions.k8s.io,resources=customresourcedefinitions,verbs=get;list;watch
+//+kubebuilder:rbac:groups=operators.coreos.com,resources=clusterserviceversions,verbs=get;list;update
+
+//+kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclusters,verbs=get;list;watch
+//+kubebuilder:rbac:groups=ceph.rook.io,resources=cephclusters,verbs=get;list;watch
+//+kubebuilder:rbac:groups=noobaa.io,resources=noobaas,verbs=get;list;watch
+//+kubebuilder:rbac:groups=monitoring.coreos.com,resources=prometheuses,verbs=get;list;watch
+//+kubebuilder:rbac:groups=monitoring.coreos.com,resources=alertmanagers,verbs=get;list;watch
+//+kubebuilder:rbac:groups=odf.ibm.com,resources=flashsystemclusters,verbs=get;list;watch
+
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.1/pkg/reconcile
+func (r *OperatorScalerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.ctx = ctx
+	r.log = log.FromContext(ctx)
+
+	r.log.Info("starting reconcile")
+
+	err := r.isOdfDependenciesCsvReady()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	err = r.addWatches(req.Name)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	err = r.scaleOperators()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	r.log.Info("ending reconcile")
+	return ctrl.Result{}, nil
+}
+
+func (r *OperatorScalerReconciler) isOdfDependenciesCsvReady() error {
+
+	odfDepsCsv := &oprv1a1.ClusterServiceVersion{}
+
+	err := r.Client.Get(r.ctx, types.NamespacedName{Name: OdfDepsSubscriptionStartingCSV, Namespace: r.OperatorNamespace}, odfDepsCsv)
+	if err != nil {
+		r.log.Error(err, "failed to get odf-deps csv", "csvName", OdfDepsSubscriptionStartingCSV)
+		return err
+	}
+
+	if odfDepsCsv.Status.Phase != oprv1a1.CSVPhaseSucceeded {
+		err = fmt.Errorf("csv %s is not in succeeded state", OdfDepsSubscriptionStartingCSV)
+		r.log.Error(err, "waiting for csv to be in succeeded state")
+		return err
+	}
+
+	return nil
+}
+
+func (r *OperatorScalerReconciler) scaleOperators() error {
+
+	var returnErr error
+
+	for kind, pkgNames := range kindToPkgNamesMapping {
+
+		crList := &metav1.PartialObjectMetadataList{}
+		crList.TypeMeta = kind.TypeMeta
+
+		err := r.Client.List(r.ctx, crList)
+		if err != nil {
+			if meta.IsNoMatchError(err) {
+				continue
+			}
+			msg := fmt.Sprintf("failed to list %s", kind.TypeMeta.Kind)
+			r.log.Error(err, msg)
+			multierr.AppendInto(&returnErr, err)
+			continue
```

Comment:
Try to minimize you use of `continue`
```suggestion
			if !meta.IsNoMatchError(err) {
				msg := fmt.Sprintf("failed to list %s", kind.TypeMeta.Kind)
				r.log.Error(err, msg)
				multierr.AppendInto(&returnErr, err)
				continue
			}
```